test/passes/optimize-instructions_all-features.wast

```diff
@@ -4021,6 +4215,40 @@
       (ref.null)
     )
   )
+  (func $optimize-boolean-context (param $x i32) (param $y i32)
+    ;; x != -1   ==>   ~x
```

Comment:
Hmm, I thought I wrote a comment for this, but I don't see it now, so writing it again: This optimization doesn't seem to happen. The output file doens't show it being changed. Maybe it was moved to another PR? I'm ok to land the test, as it is, but would be good to know what's going on.

Reply:
Yeah. Removed this